packages/core/src/Synchronizer.ts

```diff
@@ -316,6 +321,41 @@ export class Synchronizer extends EventEmitter {
         })
     }
 
+    get blocks() {
+        return this._blocks
+    }
+
+    get blockEnd() {
+        return this._blockEnd
+    }
+
+    insert(block: Promise<any>) {
```

Comment:
this function is only being used in 1 place, i think the logic would fit better in the `loadNewEvents` function

Reply:
move logic to `loadNewEvents`